### PR TITLE
SONAR-8263 isolate organizations when removing user permissions

### PR DESCRIPTION
**Pull request by simonbrandhof**

### CODE REVIEW

File: sonar-db/src/main/java/org/sonar/db/permission/AuthorizationMapper.java [hunk 1]
```diff
@@ -39,6 +39,8 @@
 
   int countRemainingUserIdsWithGlobalPermissionIfExcludeGroup(@Param("organizationUuid") String organizationUuid, @Param("permission") String permission, @Param("excludedGroupId") long excludedGroupId);
 
+  int countRemainingUsersWithGlobalPermissionIfExcludeUser(@Param("organizationUuid") String organizationUuid, @Param("permission") String permission, @Param("excludedUserId") long excludedUserId);
```
Comment (SonarTech) [inline sonar-db/src/main/java/org/sonar/db/permission/AuthorizationMapper.java]: MINOR Split this 197 characters long line (which is greater than 180 authorized). [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS00103)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* MINOR 1 minor

Watch the comments in this conversation to review them.

File: sonar-db/src/main/java/org/sonar/db/permission/GroupPermissionMapper.java [hunk 2]
```diff
@@ -34,23 +34,11 @@
...
-  @Deprecated
-  int countGroupsByPermissionQuery(@Param("query") PermissionQuery query);
+  int countGroupsByPermissionQuery(@Param("organizationUuid") String organizationUuid, @Param("query") PermissionQuery query);
 
-  /**
-   * @deprecated does not support organizations
-   */
-  @Deprecated
-  List<GroupPermissionDto> selectGroupPermissionByGroupNames(@Param("groupNames") List<String> groupNames, @Nullable @Param("projectId") Long projectId);
+  List<GroupPermissionDto> selectByGroupIds(@Param("organizationUuid") String organizationUuid, @Param("groupIds") List<Long> groupIds, @Nullable @Param("projectId") Long projectId);
```
Comment (SonarTech) [inline sonar-db/src/main/java/org/sonar/db/permission/GroupPermissionMapper.java]: MINOR Split this 182 characters long line (which is greater than 180 authorized). [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS00103)

File: sonar-db/src/main/java/org/sonar/db/permission/template/PermissionTemplateDao.java [hunk 3]
```diff
@@ -97,11 +101,19 @@ public PermissionTemplateDto selectByUuid(DbSession session, String templateUuid
...
   public List<PermissionTemplateDto> selectAll(DbSession session, String nameMatch) {
     String uppercaseNameMatch = toUppercaseSqlQuery(nameMatch);
     return mapper(session).selectAll(uppercaseNameMatch);
   }
 
+  /**
+   * @deprecated does not support organizations. Should return group ids.
+   */
+  @Deprecated
   public List<PermissionTemplateDto> selectAll(DbSession session) {
```
Comment (SonarTech) [inline sonar-db/src/main/java/org/sonar/db/permission/template/PermissionTemplateDao.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

File: sonar-db/src/main/java/org/sonar/db/permission/template/PermissionTemplateDao.java [hunk 4]
```diff
@@ -69,6 +69,10 @@ public int countUserLoginsByQueryAndTemplate(DbSession session, PermissionQuery
     return mapper(dbSession).selectUserPermissionsByTemplateIdAndUserLogins(templateId, Collections.emptyList());
   }
 
+  /**
+   * @deprecated does not support organizations. Should return group ids.
+   */
+  @Deprecated
   public List<String> selectGroupNamesByQueryAndTemplate(DbSession session, PermissionQuery query, long templateId) {
```
Comment (SonarTech) [inline sonar-db/src/main/java/org/sonar/db/permission/template/PermissionTemplateDao.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

File: sonar-db/src/main/java/org/sonar/db/permission/template/PermissionTemplateDao.java [hunk 5]
```diff
@@ -97,11 +101,19 @@ public PermissionTemplateDto selectByUuid(DbSession session, String templateUuid
     return mapper(session).selectByUuid(templateUuid);
   }
 
+  /**
+   * @deprecated does not support organizations. Should return group ids.
+   */
+  @Deprecated
   public List<PermissionTemplateDto> selectAll(DbSession session, String nameMatch) {
```
Comment (SonarTech) [inline sonar-db/src/main/java/org/sonar/db/permission/template/PermissionTemplateDao.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

**Comment by sns-seb** [comment]
error in commit title, I guess:
> SONAR-8134 improve PermissionTemplateDto
should be "SONAR-8134 improve PermissionTemplateDao", right?

File: sonar-db/src/main/java/org/sonar/db/permission/GroupPermissionDao.java [hunk 6]
```diff
@@ -51,16 +51,26 @@ public int countGroups(DbSession session, String permission, @Nullable Long comp
   }
 
   /**
+   * Returns the names of the groups that match the given query, for the given organization.
+   * The virtual group "Anyone" may be returned as the value "zero" (0L).
```
Comment (sns-seb) [inline sonar-db/src/main/java/org/sonar/db/permission/GroupPermissionDao.java]: this method returns strings. Group anyone is represented by the string "zero" or "0"?

**Comment by sns-seb** [comment]
LGTM

File: sonar-db/src/main/java/org/sonar/db/permission/GroupPermissionDao.java (hunk 6 above)
Comment (simonbrandhof) [inline sonar-db/src/main/java/org/sonar/db/permission/GroupPermissionDao.java]: wrong javadoc. Correct value is "Anyone" (`DefaultGroups.ANYONE`)

File: sonar-db/src/main/java/org/sonar/db/permission/AuthorizationMapper.java [hunk 7]
```diff
@@ -39,6 +39,8 @@
 
   int countRemainingUserIdsWithGlobalPermissionIfExcludeGroup(@Param("organizationUuid") String organizationUuid, @Param("permission") String permission, @Param("excludedGroupId") long excludedGroupId);
 
+  int countRemainingUsersWithGlobalPermissionExcludingUser(@Param("organizationUuid") String organizationUuid, @Param("permission") String permission, @Param("excludedUserId") long excludedUserId);
```
Comment (SonarTech) [inline sonar-db/src/main/java/org/sonar/db/permission/AuthorizationMapper.java]: MINOR Split this 197 characters long line (which is greater than 180 authorized). [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS00103)

**Pull request closed without merging**